python/lsst/ts/ledprojector/ledprojector_csc.py

```diff
@@ -0,0 +1,276 @@
+        config_dir: None | str = None,
+        initial_state: salobj.State = salobj.State.STANDBY,
+        override: str = "",
+        simulation_mode: bool = False,
+    ) -> None:
+        self.led_controller = None
+        self.should_be_connected = False
+
+        self.config = None
+
+        super().__init__(
+            name="LEDProjector",
+            index=0,
+            config_schema=CONFIG_SCHEMA,
+            config_dir=config_dir,
+            initial_state=initial_state,
+            override=override,
+            simulation_mode=simulation_mode,
+        )
+
+    @staticmethod
+    def get_config_pkg() -> str:
+        return "ts_config_mtcalsys"
+
+    @property
+    def controller_connected(self) -> bool:
+        """Return True if the LabJack is connected"""
+        return self.led_controller is not None and self.led_controller.connected
+
+    async def configure(self, config: Any) -> None:
+        """Configure the CSC"""
+        self.config = config
+
+    async def handle_summary_state(self) -> None:
+        """Override of the handle_summary_state function to
+        create led controller object when enabled
+
+        Raises
+        ------
+        ValueError
+            If config is None
+        """
+        self.log.info(f"handle_summary_state {salobj.State(self.summary_state).name}")
+        if self.disabled_or_enabled:
+            if self.led_controller is None:
+                if self.should_be_connected:
+                    return self.fault(
+                        code=LEDBasicState.Error,
+                        report="led controller should be connected but isn't.",
+                    )
+                else:
+                    if self.config is None:
+                        raise ValueError(
+                            "CSC tried to create LEDController with config equal to None"
+                        )
+                    self.led_controller = LEDController(
+                        config=self.config,
+                        log=self.log,
+                        simulate=self.simulation_mode,
+                    )
+            await self.connect_led()
+        else:
+            await self.disconnect_led()
+
+    async def connect_led(self) -> None:
+        """Connect to the LabJack and get status.
+        This method initiates the LEDController as well
+
+        Raises
+        ------
+        ValueError
+            If there is no LEDController object
+        asyncio.TimeoutError
+            If it takes longer than self.config.led.connect_timeout
+        """
+        await self.disconnect_led()
+        if self.led_controller is None:
+            raise ValueError("CSC Tried to use led controller without a valid object")
+
+        await asyncio.wait_for(
+            self.led_controller.connect(),
+            timeout=self.led_controller.COMMUNICATION_TIMEOUT,
+        )
+        self.should_be_connected = True
+
+    # TODO should we be catching errors
```

Comment:
Please, either remove the TODO or create a ticket for it and mention the ticket here.

Reply:
From another perspective these todos seem to be asking questions rather than an indication of something that needs doing. I would remove the todo prefix but leave the comment if stills needs answering.